### PR TITLE
c_extension: Fix clang header detection

### PR DESCRIPTION
**Pull request by ferdnyc**
Clang version N.M.O now installs files to `{libdir}/clang/N/` instead of `{libdir}/clang/N.M.O/`. Check both possibilities when scanning for headers.

(The ordering on the `itertools.product` will first check the full version under `lib`, then `lib64`, and then do the same for the major-version number.)

### CODE REVIEW

**Comment by ferdnyc** [comment]
Actually, a better way might be to just ask Clang. On my Fedora 38 system:

```console
$ clang --print-resource-dir
/usr/lib64/clang/16
```

I wonder how long that flag's been there?

**Comment by ferdnyc** [comment]
Hmm. Looks like since... [2017-ish](https://reviews.llvm.org/rG51208a38de9ea04e3e36022c02db68d911e96d1a). Well, that's not _very_ long.

(However, it _is_ now their [recommended way of discovering header paths](https://github.com/llvm/llvm-project/blob/main/clang/docs/StandardCPlusPlusModules.rst#possible-issues-failed-to-find-system-headers).)

**Comment by MathieuDuponchelle** [comment]
Thanks, this is already mergeable but do you want to add an initial check using the recommended method?

**Comment by ferdnyc** [comment]
Sorry this was delayed, see #266 for why.

I've added a `clang --print-resource-dir` run so that, if it outputs a `resource_dir` and `{resource_dir}/include` is a valid path, that'll just be returned and the `llvm-config` runs will be skipped.

File: hotdoc/extensions/c/c_extension.py [hunk 1]
```diff
@@ -70,26 +77,42 @@ def debug(message):
...
-
-    for lib in ['lib', 'lib64']:
-        p = os.path.join(prefix, lib, 'clang', version, 'include')
+    try:
+        # Clang 5.0+ can tell us directly
+        resource_dir = subprocess.check_output(
+            ['clang', '--print-resource-dir']).strip().decode()
+        include_dir = os.path.join(resource_dir, 'include')
+        if os.path.exists(include_dir):
+            return include_dir
```
Comment (ferdnyc) [inline hotdoc/extensions/c/c_extension.py:91]: Safer:

```suggestion
        if len(resource_dir) > 0:
            include_dir = os.path.join(resource_dir, 'include')
            if os.path.exists(include_dir):
                return include_dir
```

**Comment by MathieuDuponchelle** [comment]
thanks, let's go :)

**Pull request merged**